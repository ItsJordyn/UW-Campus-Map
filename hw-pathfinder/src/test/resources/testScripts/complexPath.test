# Tests the Dijkstra's algorithm for a complex graph.

CreateGraph complexGraph

AddNode complexGraph A
AddNode complexGraph B
AddNode complexGraph C
AddNode complexGraph D
AddNode complexGraph E

AddEdge complexGraph A B 3.0
AddEdge complexGraph A C 1.0
AddEdge complexGraph A D 4.0
AddEdge complexGraph B C 5.0
AddEdge complexGraph B E 10.0
AddEdge complexGraph C E 11.0
AddEdge complexGraph C A 9.0
AddEdge complexGraph E D 7.0
AddEdge complexGraph D C 2.0

FindPath complexGraph A E