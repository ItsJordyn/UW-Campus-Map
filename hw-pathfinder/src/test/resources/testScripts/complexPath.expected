# Tests the Dijkstra's algorithm for a complex graph.

created graph complexGraph

added node A to complexGraph
added node B to complexGraph
added node C to complexGraph
added node D to complexGraph
added node E to complexGraph

added edge 3.000 from A to B in complexGraph
added edge 1.000 from A to C in complexGraph
added edge 4.000 from A to D in complexGraph
added edge 5.000 from B to C in complexGraph
added edge 10.000 from B to E in complexGraph
added edge 11.000 from C to E in complexGraph
added edge 9.000 from C to A in complexGraph
added edge 7.000 from E to D in complexGraph
added edge 2.000 from D to C in complexGraph

path from A to E:
A to C with weight 1.000
C to E with weight 11.000
total cost: 12.000